### ISSUE TITLE
DisplayServer.get_display_safe_area() wrongly accounts for taskbar on Windows, even on fullscreen mode

### ISSUE THREAD

**Issue by danijmn** [godotengine/godot#99809]
### Tested versions

- Reproducible in: 4.3.stable, 4.4.dev5

### System information

Windows 10.0.19045 - Multi-window, 1 monitor - Vulkan (Forward+) - dedicated GeForce GTX 765M - Intel(R) Core(TM) i7-4700MQ CPU @ 2.40GHz (8 threads)

### Issue description

On a Windows 10 laptop with a 1920x1080 pixel screen with NO notches/cutouts, both DisplayServer.get_display_safe_area() and DisplayServer.screen_get_usable_rect() report a usable height of only 1040 pixels.

This seems to suggest that the Windows taskbar is being accounted for (it is exactly 40 pixels), which is neither useful nor a true account of the screen's safe area, especially for a fullscreen application. Running the application in windowed mode, fullscreen mode or exclusive fullscreen mode makes no difference to the reported value.

![Image](https://github.com/user-attachments/assets/af5dcb0c-63c9-4de6-ad5d-f7bf06a60813)

### Steps to reproduce

Import the MRP and run the example scene.
Verify that toggling between exclusive fullscreen mode and windowed mode (with the "T" key) does not change the reported safe_area / usable_rect values.

### Minimal reproduction project (MRP)

[wrong_safe_area.zip](https://github.com/user-attachments/files/17952301/wrong_safe_area.zip)

**Comment by danijmn** [comment]
Just confirmed that the issue persists on Windows 11. In a laptop with a 1920x1080 pixel screen (no notches/cutouts), Godot reports a "safe" height of 1020 pixels (the taskbar is 60 pixels high).

System info: Godot v4.3.stable - Windows 10.0.26100 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4060 Laptop GPU (NVIDIA; 32.0.15.6103) - AMD Ryzen 7 7840HS w/ Radeon 780M Graphics (16 Threads)

**Comment by Wusiwbsiw** [comment]
I confirmed the issue on my Win11 laptop with a 2560*1600 pixel screen,Godot reports a 2560*1528. I tried to enable the hidden taskbar or change the Zoom ratio of diplay but nothing happened.
System info:Godot v4.4 dev 5 - （Windows 11 26100.3476）（NVIDIA GeForce RTX 4060 Laptop GPU ）（AMD Ryzen 7 8845H w/ Radeon 780M Graphics  3.80 GHz）

**Comment by bruvzg** [comment]
> This seems to suggest that the Windows taskbar is being accounted for

Not sure what you expected it to return. `get_display_safe_area` is documented to fall back to `screen_get_usable_rect` on desktop platforms:

> **Note:** Currently only implemented on Android and iOS. On other platforms, `screen_get_usable_rect(SCREEN_OF_MAIN_WINDOW)` will be returned as a fallback. See also `screen_get_usable_rect`.

And `screen_get_usable_rect` is accounting for taskbar (or macOS menu bar and other similar elements):

> Returns the portion of the screen that is not obstructed by a status bar in pixels.

**Comment by danijmn** [comment]
Yes, it's documented now since 4.4, but [wasn't documented in 4.3](https://docs.godotengine.org/en/4.3/classes/class_displayserver.html#class-displayserver-method-get-display-safe-area) which was the latest version when the issue was opened. I agree it's less of an issue now that this behaviour is documented.

Anyway, I would expect `get_display_safe_area` to account for notches and screen cutouts, exactly as you would expect on a phone. And I wouldn't expect it to account for non-permanent virtual elements like the Windows taskbar, which does not impact the screen's visible area when the game is running fullscreen. `screen_get_usable_rect` works as expected because the status bar behaviour is documented, though maybe the documentation should mention that it continues to report the status bar even in fullscreen.

I don't have a macbook pro with a notch to test `get_display_safe_area` on, but I expect it should return the correct value because, in that particular case, the menu bar is a true permanent obstacle that must be accounted for even in fullscreen mode. AFAIK macOS even blacks out the entire horizontal area to the left and to the right of the notch when running a fullscreen app, which means that from the application's point of view it might as well just be running on a smaller screen - though I don't know if macOS automatically reports a smaller `screen_get_size` rect in that case.

**TLDR:** possible divergent behaviour in different platforms and when running windowed vs fullscreen, difficult to know how to handle safe areas consistently. So far very few laptops have a notch, so this shouldn't be an issue. Whether that will hold true in the future, who knows. Therefore, I leave it up to your judgment if this issue should be closed.
